amandaguan-ag/todo
# Send task email notification

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓  | :sparkles: New feature     |
|     | :bug: Bugfix               |
|     | :hammer: Refactoring       |
|    | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Acceptance Criteria

- [x] Users can enter their name and email to receive task summaries via email.
- [x] Email notifications are sent summarizing tasks due within the next 7 days.
- [x] A visually appealing email template is used to summarize tasks, with names, due dates, and priority levels.

## Related Issue

Closes #8 

## Description

This PR introduces a feature that sends weekly email notifications to users summarizing their tasks due within the next 7 days, providing them with an overview without needing to manually check the application.

### Testing Steps / QA Criteria

To prepare for testing, please ensure you've followed the [setup steps](https://github.com/amandaguan-ag/todo#setup) outlined in the project's README, including running any necessary database migrations to update your local database schema.

1. Checkout to this branch with `git checkout 4.1-email-notification-2`.
2. Make sure you are in the backend directory and run the database migrations to update your schema by executing `npm run typeorm migration:run`.
3. Start the backend service by running `npm run start` in the backend directory.
4. In a separate terminal, navigate to the frontend directory and run `npm start` to launch the frontend.
5. Navigate to [http://localhost:3000](http://localhost:3000) to access the Todo List Application.
6. Test the user entry form by adding a new user's name and email.
7. Add new tasks for a user and set due dates within the next 7 days.
8. Adjust the Cron time to a few minutes later to simulate the cron job and test the email notifications.

## TODOs

- Finalize styling of the email template.
- Implement tests for the `NotificationService`.
